pkg/jsonmerge/object.go

```diff
@@ -0,0 +1,73 @@
+package jsonmerge
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+
+	"dario.cat/mergo"
+)
+
+type objectMerger struct {
+	io.Writer
+	dst map[string]interface{}
+}
+
+// NewObjectMerger creates a Merger for JSON objects.
+func NewObjectMerger(w io.Writer) Merger {
+	return &objectMerger{
+		Writer: w,
+		dst:    make(map[string]interface{}),
+	}
+}
+
+func (merger *objectMerger) NewPage(r io.Reader, isLastPage bool) io.ReadCloser {
+	return &objectMergerPage{
+		merger: merger,
+		Reader: r,
+	}
+}
+
+func (merger *objectMerger) Close() error {
+	// Marshal to JSON and write to output.
+	buf, err := json.Marshal(merger.dst)
+	if err != nil {
+		return err
+	}
+
+	_, err = merger.Writer.Write(buf)
+	return err
+}
+
+type objectMergerPage struct {
+	merger *objectMerger
+
+	io.Reader
+	buffer bytes.Buffer
+}
+
+func (page *objectMergerPage) Read(p []byte) (int, error) {
```

Comment:
I think a doc comment here would be quite useful along the lines of:

"The `Read` function will never write into the provided buffer, it exists to copy from the page reader into the page buffer for later unmarshaling"

I went around a bit in circles wondering why we were always returning `0`.

Reply:
I will add a comment, but keep it mind it won't generate a doc comment. `objectMergerPage` is private. All you'd get for `Read` is from `io.ReadCloser.Read`.